plugins/xlsxreader.cpp

```diff
@@ -114,6 +116,8 @@ int get_row(lua_State *L) {
         auto cells = std::vector<std::string>();
         while (get_next_cell(sheet_handle, value)) {
             cells.push_back(value);
+            if (max_tokens > 0 && cells.size() >= max_tokens)
+                break;
```

Comment:
I'm seeing an interesting issue (I think it's an issue, not 100% sure) where passing `max_tokens <= num_cells_in_row` results in every other call to `get_row()` returning an empty row.

Specifically, I have a sheet with 8 columns (of irrelevant data). Calling `sheet:get_row(8)` returns a table with 8 elements, then a table with 0, then a table of 8, and so on. `sheet:get_row(3) behaves similarly, but `sheet:get_row(9)` always returns a table with 8 elements. I've compared to the data that's in the spreadsheet and confirmed that it isn't skipping any rows - it's just returning empty rows for every other call in the cases I've described.

Is this something you can reproduce on your end? I can attach a test file if not. My suspicion is that this might be going unnoticed if quickfort simply ignores empty rows, but it's worth looking into just in case other scripts end up using this feature.

Reply:
As far as I can tell, this is an issue upstream, even as of https://github.com/brechtsanders/xlsxio/commit/e3acace39e5fb153f5ce3500a4952c2bf93175bd

This patch seems to fix the issue, although it is a little less efficient when `max_tokens < num_cells_in_row`:
```diff
diff --git a/plugins/xlsxreader.cpp b/plugins/xlsxreader.cpp
index d9a781f1..12a59113 100644
--- a/plugins/xlsxreader.cpp
+++ b/plugins/xlsxreader.cpp
@@ -115,9 +115,9 @@ int get_row(lua_State *L) {
         std::string value;
         auto cells = std::vector<std::string>();
         while (get_next_cell(sheet_handle, value)) {
-            cells.push_back(value);
-            if (max_tokens > 0 && cells.size() >= max_tokens)
-                break;
+            if (max_tokens <= 0 || cells.size() <= max_tokens) {
+                cells.push_back(value);
+            }
         }
         Lua::PushVector(L, cells, true);
     }
```

Reply:
Thanks for finding this! when quickfort reads less than all the cells, it's because it's scanning for blueprint modelines, so the extra empty rows don't affect anything. when the file is read for actual blueprint data, all cells are read so the issue doesn't manifest. I'll use your fix and add an explanatory comment.